Invalid relative name failure when compiling

When building gwt-resources (on Windows) the compile fails because of invalid relative paths:

```
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: Invalid relative name: org\gwtproject\resources\client/hello.txt
    at com.sun.tools.javac.main.Main.compile (Main.java:553)
    at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
    at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
    at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:126)
    at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:174)
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:943)
    at org.apache.maven.plugin.compiler.TestCompilerMojo.execute (TestCompilerMojo.java:164)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Seems that the APT Filer doesn't accept backslashes for relative paths. I will make a PR with a fix that I have already made. Someone with a Unix based system should test this fix before merging to check if this works for both systems.

--- a/core/src/main/java/org/gwtproject/resources/rg/resource/impl/ResourceOracleImpl.java
+++ b/core/src/main/java/org/gwtproject/resources/rg/resource/impl/ResourceOracleImpl.java
@@ -215,7 +215,7 @@ public class ResourceOracleImpl implements ResourceOracle {
     for (Location location : searchLocations) {
       String path = "";
       if (pkg.length() > 0) {
-        path = String.valueOf(pkg).replace('.', File.separatorChar) + '/';
+        path = String.valueOf(pkg).replace('.', '/') + '/';
       }
       try {
         FileObject fileObject = aptContext.filer.getResource(location, "", path + relativeName);

